.pre-commit-config.yaml

```diff
@@ -145,6 +116,13 @@ repos:
         entry: bash -c 'pnpm exec vue-tsc -p frontend --noEmit'
         language: system
         pass_filenames: false
+      - id: eslint
```

Comment:
Can we do the same for prettier?

Reply:
I think prettier is a different case. If you check out the original discussion and Dhruv and I had that led to this change, we focused on the fact that ESLint only runs if you're making changes to JS files. And if you're making changes to JS files, the thinking is, then you should have run `pnpm install` (to run unit tests, try the changes out locally, etc). Therefore, ESLint can reliably run as a local hook because the times when it does run are times when the local environment is expected to have `pnpm install` run before committing. Types follow the same logic, I believe, and also wouldn't work without `pnpm install` having been run beforehand.

Prettier, on the other hand, we rely on to format our Markdown files, which includes, for example, the catalog DAG docs, the documentation site, etc. So while the dependencies duplication is annoying for prettier as well, I don't know that there's much we can do to fix it, at least not the same way we can for ESLint, because the conditions under which prettier is depended on do not necessitate `pnpm install`.

We could make `pnpm install`, at least for the workspace root. If there's a way to tell `pnpm` to only install workspace root dependencies, maybe that is a workable assumption, especially if it simplifies our configuration.

In any case, it should be a separate issue. The considerations that went into making this possible for ESLint are different from the ones we need to think about for prettier.

Reply:
I see, thank you for explanation!